collector/client.go

```diff
@@ -93,6 +94,49 @@ func (c *CacheObjectClient) GetCounters() (types.Counters, error) {
 	return counters, err
 }
 
+/*GetServiceTimes fetches service times from squid cache manager */
+func (c *CacheObjectClient) GetServiceTimes() (types.Counters, error) {
+	conn, err := connect(c.hostname, c.port)
+
+	if err != nil {
+		return types.Counters{}, err
+	}
+
+	r, err := get(conn, "service_times", c.basicAuthString)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != 200 {
+		return nil, fmt.Errorf("Non success code %d while fetching service_times metrics", r.StatusCode)
+	}
+
```

Comment:
Since now we have two places to connect to the squid server, I think it's better if we extract this part into its own separate function with the path (ie. `service_time`) as a parameter, wdyt?

Reply:
yes this makes total sense. Update 👍